src/php/ext/grpc/channel.c

```diff
@@ -146,14 +175,83 @@ PHP_METHOD(Channel, __construct) {
       php_grpc_zend_hash_del(array_hash, "credentials", sizeof("credentials"));
     }
   }
-  php_grpc_read_args_array(args_array, &args TSRMLS_CC);
-  if (creds == NULL) {
-    channel->wrapped = grpc_insecure_channel_create(target, &args, NULL);
+  if (php_grpc_zend_hash_find(array_hash, "force_new", sizeof("force_new"),
+                              (void **)&force_new_obj) == SUCCESS) {
+    if (PHP_GRPC_BVAL_IS_TRUE(force_new_obj)) {
+      force_new = true;
+    }
+    php_grpc_zend_hash_del(array_hash, "force_new", sizeof("force_new"));
+  }
+
+  // parse the rest of the channel args array
+  if (php_grpc_read_args_array(args_array, &args TSRMLS_CC) == FAILURE) {
+    return;
+  }
+
+  // Construct a hashkey for the persistent channel
+  // Currently, the hashkey contains 3 parts:
+  // 1. hostname
+  // 2. hash value of the channel args array (excluding "credentials"
+  //    and "force_new")
+  // 3. (optional) hash value of the ChannelCredentials object
+  //
+  // Note that, if the ChannelCredentials object was created by
+  // "composing" a ChannelCredentials and CallCredentials object
+  // together, the CallCredentials portion will be ignored, when
+  // constructing this hashkey.
+  //
+  // In other words, the same ChannelCredentials object, composed
+  // with 2 different CallCredentials objects, will return the same
+  // underlying grpc_channel. So proceed with caution.
+
+  php_serialize_data_t var_hash;
+  smart_str buf = {0};
+  PHP_VAR_SERIALIZE_INIT(var_hash);
+  PHP_GRPC_VAR_SERIALIZE(&buf, args_array, &var_hash);
+  PHP_VAR_SERIALIZE_DESTROY(var_hash);
+
+  char sha1str[41];
+  generate_sha1_str(sha1str, PHP_GRPC_SERIALIZED_BUF_STR(buf),
+                    PHP_GRPC_SERIALIZED_BUF_LEN(buf));
+
+  php_grpc_int key_len = target_length + 40;
+  if (creds != NULL && creds->hashstr != NULL) {
+    key_len += 40;
+  }
+  char *key = emalloc(key_len);
```

Comment:
It's hidden in each object's PHP's destructor: https://github.com/grpc/grpc/blob/master/src/php/ext/grpc/php7_wrapper.h#L56